src/acceleration/test/AccelerationMasterSlaveTest.cpp

```diff
@@ -416,70 +417,70 @@ BOOST_AUTO_TEST_CASE(testVIQNIMVJpp)
-    BOOST_TEST(testing::equals((*data.at(0)->values)(1), 1.05000000000000004441e+00), (*data.at(0)->values)(1));
-    BOOST_TEST(testing::equals((*data.at(0)->values)(2), 1.06000000000000005329e+00), (*data.at(0)->values)(2));
-    BOOST_TEST(testing::equals((*data.at(0)->values)(3), 1.07000000000000006217e+00), (*data.at(0)->values)(3));
-    BOOST_TEST(testing::equals((*data.at(1)->values)(0), 1.99000000000000010214e-01), (*data.at(1)->values)(0));
-    BOOST_TEST(testing::equals((*data.at(1)->values)(1), 1.99000000000000010214e-01), (*data.at(1)->values)(1));
-    BOOST_TEST(testing::equals((*data.at(1)->values)(2), 1.99000000000000010214e-01), (*data.at(1)->values)(2));
-    BOOST_TEST(testing::equals((*data.at(1)->values)(3), 1.99000000000000010214e-01), (*data.at(1)->values)(3));
+    BOOST_TEST(testing::equals(data.at(0)->values()(0), 1.04000000000000003553e+00), data.at(0)->values()(0));
+    BOOST_TEST(testing::equals(data.at(0)->values()(1), 1.05000000000000004441e+00), data.at(0)->values()(1));
+    BOOST_TEST(testing::equals(data.at(0)->values()(2), 1.06000000000000005329e+00), data.at(0)->values()(2));
+    BOOST_TEST(testing::equals(data.at(0)->values()(3), 1.07000000000000006217e+00), data.at(0)->values()(3));
+    BOOST_TEST(testing::equals(data.at(1)->values()(0), 1.99000000000000010214e-01), data.at(1)->values()(0));
+    BOOST_TEST(testing::equals(data.at(1)->values()(1), 1.99000000000000010214e-01), data.at(1)->values()(1));
+    BOOST_TEST(testing::equals(data.at(1)->values()(2), 1.99000000000000010214e-01), data.at(1)->values()(2));
+    BOOST_TEST(testing::equals(data.at(1)->values()(3), 1.99000000000000010214e-01), data.at(1)->values()(3));
     utils::append(newdvalues, 10.0);
     utils::append(newdvalues, 10.0);
     utils::append(newdvalues, 10.0);
     utils::append(newdvalues, 10.0);
   } else if (context.isRank(2)) { //Slave2
     // empty proc
   } else if (context.isRank(3)) { //Slave3
-    BOOST_TEST(testing::equals((*data.at(0)->values)(0), 1.00000000000000000000e+00), (*data.at(0)->values)(0));
-    BOOST_TEST(testing::equals((*data.at(0)->values)(1), 1.01000000000000000888e+00), (*data.at(0)->values)(1));
-    BOOST_TEST(testing::equals((*data.at(1)->values)(0), 1.99000000000000010214e-01), (*data.at(1)->values)(0));
-    BOOST_TEST(testing::equals((*data.at(1)->values)(1), 1.99000000000000010214e-01), (*data.at(1)->values)(1));
+    BOOST_TEST(testing::equals(data.at(0)->values()(0), 1.00000000000000000000e+00), data.at(0)->values()(0));
+    BOOST_TEST(testing::equals(data.at(0)->values()(1), 1.01000000000000000888e+00), data.at(0)->values()(1));
+    BOOST_TEST(testing::equals(data.at(1)->values()(0), 1.99000000000000010214e-01), data.at(1)->values()(0));
+    BOOST_TEST(testing::equals(data.at(1)->values()(1), 1.99000000000000010214e-01), data.at(1)->values()(1));
     utils::append(newdvalues, 10.0);
     utils::append(newdvalues, 10.0);
   }
 
-  data.begin()->second->values = &newdvalues;
```

Comment:
This use-case is removed by this PR. One cannot anymore just provide new data and replace the pointer. Instead, one can only overwrite the data `CouplingData` is pointing to or one has to provide a new `PtrData`.

From my point of view it is good to remove this use-case, since one can otherwise easily introduce data that is not consistent with the other members of `CouplingData` (dimension...).